index.js

```diff
@@ -3,11 +3,26 @@ module.exports = function (str, opts) {
 	str = str.toString();
 	opts = opts || {};
 
-	var preserve = !opts.all;
+	var comment = '';
 	var currentChar = '';
 	var insideString = false;
+	var preserveFilter;
+	var preserveImportant = (opts.preserve === false || opts.all === true) ? false : true;
 	var ret = '';
 
+	//use preserveFilter as an interface for opts.preserve function and RegExp values
+	if (opts !== undefined && opts.preserve !== undefined) {
+		if ((typeof opts.preserve).toLowerCase() === 'function') {
```

Comment:
True. Time for me to dispense with my paranoia that someday `typeof` will return a string in a case other than lower case.